operator/controllers/common/providers/keptnmetric/keptnmetric.go

```diff
@@ -32,3 +31,24 @@ func (p *KeptnMetricProvider) EvaluateQuery(ctx context.Context, objective klcv1
 
 	return metric.Status.Value, metric.Status.RawValue, nil
 }
+
+func (p *KeptnMetricProvider) GetKeptnMetric(ctx context.Context, objective klcv1alpha3.Objective, namespace string) (*metricsapi.KeptnMetric, error) {
+	metric := &metricsapi.KeptnMetric{}
+
+	if objective.KeptnMetricRef.Namespace != "" {
+		if err := p.K8sClient.Get(ctx, types.NamespacedName{Name: objective.KeptnMetricRef.Name, Namespace: objective.KeptnMetricRef.Namespace}, metric); err != nil {
+			p.Log.Error(err, "Failed to get KeptnMetric from objective namespace")
+			return nil, err
+		}
+	} else {
+		if err := p.K8sClient.Get(ctx, types.NamespacedName{Name: objective.KeptnMetricRef.Name, Namespace: namespace}, metric); err != nil {
+			p.Log.Error(err, "Failed to get KeptnMetric from KeptnEvaluation resource namespace")
+			if err := p.K8sClient.Get(ctx, types.NamespacedName{Name: objective.KeptnMetricRef.Name, Namespace: "keptn-lifecycle-toolkit-system"}, metric); err != nil {
```

Comment:
please use a constant for the `keptn-lifecycle-toolkit-system`

Reply:
resolved